src/interceptors/ClientRequest/utils/cloneIncomingMessage.test.ts

```diff
@@ -1,20 +1,25 @@
 import { Socket } from 'net'
 import { IncomingMessage } from 'http'
+import { Stream, Readable, EventEmitter } from 'stream'
 import { cloneIncomingMessage, IS_CLONE } from './cloneIncomingMessage'
 
 test('clones a given IncomingMessage', () => {
-  const source = new IncomingMessage(new Socket())
-  source.statusCode = 200
-  source.statusMessage = 'OK'
-  source.headers = { 'x-powered-by': 'msw' }
-  const clone = cloneIncomingMessage(source)
+  const message = new IncomingMessage(new Socket())
+  message.statusCode = 200
+  message.statusMessage = 'OK'
+  message.headers = { 'x-powered-by': 'msw' }
+  const clone = cloneIncomingMessage(message)
+
+  // Prototypes must be preserved.
+  expect(clone).toBeInstanceOf(IncomingMessage)
+  expect(clone).toBeInstanceOf(EventEmitter)
+  expect(clone).toBeInstanceOf(Stream)
+  expect(clone).toBeInstanceOf(Readable)
```

Comment:
I'm adding assertions to check that the cloned message indeed extends all the prototypes of the original message.